### ISSUE TITLE
Rename of a remote

### ISSUE THREAD

**Issue by Pebowski** [unfoldedcircle/bug-reports#188]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

I named one of my remotes incorrectly Deion in the Configurator. I see this on the remote, go to Webconfigurator an renamed it to Denon. After restarting the remote it was also Denon in the remote, but after a while it changed back to Deion on the remote. In the Webconfigurator it is still Denon. How can I make Denon permanent on the remote?

### How to Reproduce

1. Enter the name of the remote incorrectly, i.e Deion
2. Use it in a activity
3. Correct the name in Configurator, i.e. Denon
4. Restart the remote, name is correct: Denon
5. Start the activity
6. Name reverts to Deion on the remote, in Configurator it's OK with Denon

### Expected behavior

The name on the remote should stay to Denon

### System version

1.4.5

### What part of the system affected by the problem?

UI, Web Configurator

**Comment by zehnm** [comment]
I'm trying to reproduce this issue but so far I'm not able to.

@Pebowski some questions:
- Step 4) Do you mean restarting the physical remote device?
  Is that for any particular reason, or does this issue only show after you restart the device?
- Step 6) Where does the remote name revert to the old name?
  Did you add the remote entitiy to a profile page and it changed there? Something like:

<img width="268" alt="image" src="https://github.com/unfoldedcircle/feature-and-bug-tracker/assets/17356629/b103962b-2376-4945-9275-7d10089d9f3c">

**Comment by Pebowski** [comment]
4) Yes, restarting the physical remote. It is done to correct the name on the physical remote
6) Yes, it is added to the profile page and changed there. But only on the physical remote. In Configurator all is OK. So I should add 1a) Create an Activity and add activity and remote to Profile

PS: On the physical remote the "Rename" function does not work.

**Comment by zehnm** [comment]
Found the issue: this occurs for non-english language settings. The remote UI still shows the original English name, whereas the web-configurator shows the language specific text.

How to reproduce:
1. Set language to another language than English, for example German
2. Create a remote-entity with the default name of a selected manufacturer IR dataset, e.g. `Generic TV 2 Smart Home`
3. Rename remote-entity, e.g. to `LG TV`
4. --> web-configurator shows correct name
5. --> remote-ui still shows original (English) name
6. Access the remote-entity with the REST-API to show all language strings.
   Simplified data structure:
```json
    {
        "entity_id": "uc.main.d899d6e4-ba7f-42dd-a586-1ea77b5bd8e3",
        "entity_type": "remote",
        "integration_id": "uc.main",
        "name": {
            "de": "LG TV",
            "en": "Generic TV 2 Smart Home"
        },
        "icon": "uc:remote"
    }
```

Issues:
- remote UI doesn't properly check language specific strings: no fallback to main language without a country variant. For example, it only checks `de_CH` instead of also `de` if not found.
- web-configurator only stores language specific texts with language code without country variant. For example `de` instead of `de_CH`.

We'll fix this!

**Comment by Pebowski** [comment]
Thank you, Markus. Yes, I have language German.

**Comment by Pebowski** [comment]
Also when you try to rename an activity on the physical remote, nothing happens, the old name remains. Maybe for the same reason.

**Comment by heimspielvier** [comment]
Just wanted to confirm that I am seeing the same issue when renaming activities with the language set to German.
I can also confirm that renaming an activity directly on the remote does not work either (not sure if this is related to the same root cause).

**Comment by kennymc-c** [comment]
Unfortunately, even with the current firmware 1.7.5, there are still problems with incorrect resp. old names, although https://github.com/unfoldedcircle/remote-ui/issues/7 has already been closed.
The names of the English strings are displayed on the remote after a firmware update. I have noticed this since the last 3-4 updates. By restarting the remote or the remote UI via API, the names of the selected language are displayed again.
Because renaming only changes the string of the currently selected language both in the web configurator and via the remote UI, the English string remains unchanged. This seems to affect not only normal entities but also activities and macros.
I also don't really understand why there are strings for languages here at all. Shouldn't the name always be the same regardless of the language? This doesn't seem to be the case with profile groups. Here there is only one fixed name. Wouldn't that be a better solution for other elements too?

**Comment by Pebowski** [comment]
And also the rename of an entity or activity direct on the remote still does not work, when language is German.

**Comment by kennymc-c** [comment]
On the remote? Because it works with the web configurator and shows the new name at least right after you rename it. If the remote shows the old name it's probably the English name that has not been changed by the rename function

**Comment by Pebowski** [comment]
Yes, directly on the remote, when you long press on a name and then choose "Rename"

**Comment by kennymc-c** [comment]
Mentioned again in #568 that it will be fixed. Hopfully we don't have to wait another 20 months...

**Comment by zehnm** [comment]
Fixed with #568 in 2.6.6

The Remote user interface should dynamically update most changes made in the web-configurator. There are a few sreens in the UI left that are only refreshed if the screen is re-opened. For example the integration overview in settings.
Please open a new issue if texts are still out of sync between the web-configurator and the Remote after it has been restarted.
